### ISSUE TITLE
Cannot read property 'name' of undefined when merge requried and not

### ISSUE THREAD

**Issue by Fi1osof** [Urigo/merge-graphql-schemas#159]
Bug.

merge-graphql-schemas@1.5.3

Merge two types:
```
input UserGroupUpdateInput {
  name: String
  Users: UserUpdateManyWithoutGroupsInput
}
```
```
input UserGroupUpdateInput {
  name: String!
}
```
(In first name not required, in second required).
Got error: 
```
...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4312
      if (field.type.type.name.value !== original.type.type.name.value) {
                                                            ^

TypeError: Cannot read property 'name' of undefined
    at ...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4312:61
    at Array.reduce (<anonymous>)
    at _makeMergedFieldDefinitions (...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4301:46)
    at defs.filter.reduce.Query (...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4345:17)
    at Array.reduce (<anonymous>)
    at _makeMergedDefinitions (...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4331:6)
    at mergeTypes (...server/node_modules/merge-graphql-schemas/dist/index.cjs.js:4384:20)
    at CoreModule.getApiSchema (...server/src/node_modules/prisma-module/src/index.js:112:13)
    at CoreModule.getApiSchema (...server/src/modules/index.js:125:5)
    at generateSchema (...server/src/node_modules/react-cms-schema/src/index.js:48:27)

```

**Comment by cfnelson** [comment]
@Fi1osof Thanks for reporting this, the intended behaviour in this case would be to throw an error alerting the user to conflicting types in the two files. Guessing this might have been missed for  `input` types? A PR would be welcome.

**Comment by Fi1osof** [comment]
@cfnelson i think need not error alerting, just make result field required.

**Comment by cfnelson** [comment]
@Fi1osof  I think if there are any conflicts like the above example, they should be manually fixed, or a specific object is passed in as an option on how to resolve the multiple conflicting  field type definitions into the specific type you specified. 

e.g -> Something like below could be passed into the mergeTypes function as an option to resolve your original example. It would instruction the mergeTypes on which type field definintion to resolve/merge. This is assuming someone doesn't wish to manually fix their graphql.

```js
const resolveConflicts = {
  UserGroupUpdateInput: "String!"
};
```

**Comment by Fi1osof** [comment]
@cfnelson graphql allow set notrequired variable as required. For example:
```
input Where: String

query test($where: String!){
    docs(where: $where)
}
```
But can not like
```
input Where: String!

query test($where: String){
    docs(where: $where)
}
```
So via merge schema have ability set notrequired field as required.
input Where: String
input Where: String!
became input Where: String!

**Comment by kamilkisiela** [comment]
Should be fixed by https://github.com/ardatan/graphql-toolkit/pull/189

**Comment by kamilkisiela** [comment]
Fixed with `v1.7.0`
